jekyll/_cci2/build-publish-snap-packages.md

```diff
@@ -0,0 +1,176 @@
+          command: |
+            mkdir .snapcraft
+            echo $SNAPCRAFT_LOGIN_FILE | base64 --decode --ignore-garbage > .snapcraft/snapcraft.cfg
+            snapcraft push *.snap --release stable
+...
+```
+
+In this example, Snapcraft automatically looks for login credentials in `.snapcraft/snapcraft.cfg` so we decode the environment variable we made previously right into that location. `snapcraft push` is then used to upload the .snap file into the Snap Store.
+
+### Uploading vs Releasing
+
+`snapcraft push *.snap` by default will upload the snap to the Snap Store, run any store checks on the server side, and then stop. The snap won't be "released" meaning users won't automatically see the update. The snap can be published locally with the `snap release <release-id>` command or by logging into the Snap Store and clicking the release button.
+
+In typical CircleCI fashion, we can go fully automated (as in the above example) but using the `--release <channel>` flag. This uploads the snap, does Store side verification, and then will automatically release the snap in the specified channels. 
+
+
+## Workflows
+
+We can utilize multiple jobs to better organize our snap build. A job to build/compile the actual project, a job to build the snap itself, and a job that published the snap (and other packages) only on `master` would all be useful.
+
+[Workflows][workflows] can help with building snaps in two ways:
+
+1. **Snap Store Channels** - As we mentioned in the previous section, when we upload to the Store we could optionally release at the same time. This allows us to designate specific jobs on CircleCI to deploy to specific Snap Channels. For example, the `master` branch could be used to deploy to the `edge` channel` while tagged releases could be used to deploy to the `stable` channel.
+1. **Parallelize Packing** - If your software is being packaged as a snap as well as something else, say a flatpak, .deb, .apk, etc, each package type could be placed in its own job and all run parallel. This allows your build to complete must fast than if the .deb package could start to build until the snap completed, and so on.
+
+Utilize CircleCI `workspaces` to move a generated snap file between jobs when neccessarily. Here's an example showing a snippet from the "from" job and a snippet of the "to" job:
+
+```
+... # from a job that already has the snap
+      - persist_to_workspace:
+          root: .
+          paths:
+            - "*.snap"
+... # to the next job that needs the snap
+      - attach_workspace:
+          at: .
+...
+```
+
+Below is a complete example of how a snap package could be built on CircleCI. This same process is used the build the Snap pakcage for the [CircleCI Local CLI][local-cli-repo].
+
+
+## Full Example Config
+
+```
+workflows:
+  version: 2
+  main:
+    jobs:
+      - build
+      - publish:
+          requires:
+            - build
+          filters:
+            branches:
+              only: master
```

Comment:
Our standard has been to keep the `workflows` section at the bottom. Any reason you've got it at the top?

Reply:
It's confusing otherwise. More and more examples, including in the blog and Connor's recent refactor of the main website has Workflows at the top. When you have multiple jobs, you have absolutely no context in how many there are, how they relate to each other, and when they should run until you scroll through potentially tons of lines to see the Workflow section.

Often you can't even know if Workflows are involved into you scroll all the way to the bottom.

When this happens to me I get to the bottom, see that there's Workflows, and then to have to go back up now and put things into context.

I'd go as far as to say our standard everywhere should be Workflows config first.

Reply:
We put it at the bottom because it was confusing for people migrating to have it at the top for multiple reasons (one of them being the version number), based on comments and feedback from our users and developers.

Reply:
Makes sense, although we should probably reevaluate this as the 1.0 audience shrinks. I really like @felicianotech's points about seeing the high-level with workflows before delving into specific jobs.

Reply:
So while I still think the top is better for everyone, which way are we deciding for this doc? I'll move it to the bottom to speed this up ATM.

Reply:
bottom please, thanks!

Reply:
Done